test/oauth-server/service/authorize.go

```diff
@@ -68,7 +68,10 @@ func (requestHandler *RequestHandler) authorize(w http.ResponseWriter, r *http.R
 		</html>`
 		w.WriteHeader(http.StatusOK)
 		w.Header().Set(contentTypeHeaderKey, "text/html;charset=UTF-8")
-		w.Write([]byte(body))
+		_, err = w.Write([]byte(body))
+		if err != nil {
+			writeError(w, err, http.StatusBadRequest)
```

Comment:
when `w.Write` fails you can only log error.  it is not recommended to call twice `w.WriteHeader`